### PR TITLE
Remove UbuntuDesktopInstallerApp dependencies from individual pages

### PR DESCRIPTION
**Pull request by jpnurmi**
WSL OOBE wizard is going to reuse the `WelcomePage`, but the app will be different with another set of routes. Providing the app locale helps to eliminate the dependency to `UbuntuDesktopInstallerApp`.

Locale and theme are similar in the way that changing either rebuilds the whole app, so the proposal is to bundle them together as app-wide settings to avoid yet another tiny provider (if theme & locale were provided separately).

### CODE REVIEW

**Comment by jpnurmi** [comment]
Rebased to resolve conflicts

File: packages/ubuntu_desktop_installer/lib/settings.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  /// changes (the entire app will be rebuilt anyway).
+  static Settings of(BuildContext context, {bool listen = true}) {
+    return Provider.of<Settings>(context, listen: listen);
+  }
+
+  /// The current application theme.
+  ThemeMode get theme => _theme;
+  ThemeMode _theme = ThemeMode.system;
+
+  /// Applies a theme matching the given [brigtness].
```
Comment (robert-ancell) [inline packages/ubuntu_desktop_installer/lib/settings.dart:26]: typo in "brightness".

File: packages/ubuntu_desktop_installer/lib/main.dart [hunk 2]
```diff
@@ -9,13 +9,13 @@ import 'package:subiquity_client/subiquity_client.dart';
...
 import 'app.dart';
-import 'app_theme.dart';
 import 'disk_storage_model.dart';
 import 'keyboard_service.dart';
 import 'l10n/app_localizations.dart';
+import 'settings.dart';
 
 Future<void> main() async {
-  final themeSettings = GSettings(schemaId: 'org.gnome.desktop.interface');
+  final gsettings = GSettings(schemaId: 'org.gnome.desktop.interface');
```
Comment (robert-ancell) [inline packages/ubuntu_desktop_installer/lib/main.dart:18]: This is probably better named "interfaceSettings".

**Review by robert-ancell — APPROVED** [review]
Aside from the small nits LGTM.

**Pull request merged**